packages/block-library/src/query-no-results/index.php

```diff
@@ -33,7 +47,13 @@ function render_block_core_query_no_results( $attributes, $content, $block ) {
 		$query = $wp_query;
 	} else {
 		$query_args = build_query_vars_from_query_block( $block, $page );
-		$query      = new WP_Query( $query_args );
+
+		// Add search parameter if instant search is enabled and search query exists
+		if ( $enhanced_pagination && $instant_search_enabled && ! empty( $search_query_direct ) ) {
+			$query_args['s'] = $search_query_direct;
+		}
+
+		$query = new WP_Query( $query_args );
 	}
 
 	if ( $query->post_count > 0 ) {
```

Comment:
We can include the contents of this block but hide them when there are no posts. That way, the CSS and JS of the inner blocks will be correctly loaded and the blocks will work when the user navigates to a "no results" page in the client.

We can remove this workaround if we finally improve how client-side navigation works before WP 6.8.

Reply:
Oh, nice. That's a good idea 👍 

Can we add this in a follow-up PR though? I feel that it's not essential for the first version of the experimental block.

Reply:
Sure.